### PR TITLE
Django makemessages with xgettext parameters

### PR DESCRIPTION
**Pull request by AlexandreArpin**
First time creating a python package for django! The package structure was heavily inspired by the django-celery-fulldbresult package.

Essentially, this allows you to pass any flag to the xgettext command when it's used by django.

### CODE REVIEW

File: django-makemessages-xgettext/management/commands/xgettextmakemessages.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+                    help='Passes arguments to the xgettext command (e.g. '
+                         '--add-location=file). Use multiple times to pass '
+                         'additional arguments.'
+                    ),
+    )
+
+    def handle_noargs(self, *args, **options):
+
+        for option in options["xgettext"]:
+            self.xgettext_options = self.xgettext_options[:] + [option]
```
Comment (bartdag) [inline django-makemessages-xgettext/management/commands/xgettextmakemessages.py]: self.xgettext_options.extend plutôt que de générer une nouvelle liste + liste?

File: setup.py [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+
+setup(
+    name='django-makemessages-xgettext',
+
+    # Versions should comply with PEP440.  For a discussion on single-sourcing
+    # the version across setup.py and the project code, see
+    # https://packaging.python.org/en/latest/single_source_version.html
+    version='0.1.0',
+
+    description='Celery result backend that stores everything.',
```
Comment (bartdag) [inline setup.py]: copy paste error

File: README.rst [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+------------
+
+::
+
+    pip install django-makemessages-xgettext
+
+Usage
+-----
+
+As a way to generate .po files without the line number comment reference
```
Comment (bartdag) [inline README.rst:38]: I believe you need to add this app to your INSTALLED_APPS settings in order to work.

**Comment by bartdag** [comment]
Annnddd... you need some tests :-) You probably want to create a test project (see django fulldb result for an example).
Otherwise, I only have a few comments and it will be g2g.

@ginmrt do you mind taking a look at the pull requests?

Also @AlexandreArpin, if you want, you can create github issues to track your tasks (never refer to our internal tracking system since nobody will be able to understand/follow)

File: django-makemessages-xgettext/management/commands/xgettextmakemessages.py (hunk 1 above)
Comment (AlexandreArpin) [inline django-makemessages-xgettext/management/commands/xgettextmakemessages.py]: If you take a look at makemessages.py:

```
# Avoid messing with mutable class variables
if options.get('no_wrap'):
    self.msgmerge_options = self.msgmerge_options[:] + ['--no-wrap']
```

I couldn't figure out why, but Django does it this way. I figured it was best to stay on track.

Reference: https://github.com/django/django/blob/stable/1.7.x/django/core/management/commands/makemessages.py#L221

Comment (bartdag) [inline django-makemessages-xgettext/management/commands/xgettextmakemessages.py]: maybe it is not a mutable sequence (then that would be the only way to accomplish this).

File: README.rst [hunk 4]
```diff
@@ -0,0 +1,40 @@
+django-makemessages-xgettext - Generates translation with additional options
+============================================================================
+
+:Authors:
+  Resulto Developpement Web Inc.
+:Version: 0.1
+
+This projects has one goal:
```
Comment (ginmrt) [inline README.rst]: project

File: setup.py [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+        #   3 - Alpha
+        #   4 - Beta
+        #   5 - Production/Stable
+        'Development Status :: 4 - Beta',
+
+        # Indicate who your project is intended for
+        'Intended Audience :: Developers',
+        'Topic :: Communications',
+        'Topic :: Software Development :: Libraries :: Python Modules',
+        'Topic :: System :: Distributed Computing',
```
Comment (bartdag) [inline setup.py]: no

File: setup.py [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+    classifiers=[
+        # How mature is this project? Common values are
+        #   3 - Alpha
+        #   4 - Beta
+        #   5 - Production/Stable
+        'Development Status :: 4 - Beta',
+
+        # Indicate who your project is intended for
+        'Intended Audience :: Developers',
+        'Topic :: Communications',
```
Comment (bartdag) [inline setup.py]: no

**Comment by bartdag** [comment]
some small changes and then, g2g!

File: setup.py (hunk 5 above)
Comment (AlexandreArpin) [inline setup.py]: Hahaha how did I miss that x)

**Pull request merged**